### ISSUE TITLE
console.table doesn't properly account for certain characters

### ISSUE THREAD

**Issue by qodesmith** [oven-sh/bun#9528]
### What version of Bun is running?

1.0.33

### What platform is your computer?

Darwin 23.3.0 arm64 arm

### What steps can reproduce the bug?

```
console.table(['Hello world', `＂demo＂`])
```
☝️ Those are _not_ regular double-quote characters.

### What is the expected behavior?

The table borders should be aligned.

### What do you see instead?

The table borders are not aligned.
<img width="374" alt="Screenshot 2024-03-20 at 7 32 08 AM" src="https://github.com/oven-sh/bun/assets/11034792/78ff7182-25df-49d8-9331-67e7475bce5c">

**Comment by truongan07** [comment]
I use console.table to display document and :smile: 
- On Chromium 
![image](https://github.com/oven-sh/bun/assets/141234261/ed96b23d-a2c1-40d4-81c3-18286e75496a)

- On Terminal
![image](https://github.com/oven-sh/bun/assets/141234261/8f605984-466f-4329-901f-4c7b68683606)

![image](https://github.com/oven-sh/bun/assets/141234261/400bc5e0-6635-4dfd-b255-b49360b18be3)

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.7
```
~/Desktop/ok via 🍞 v1.1.34
❯ bun-1.1.6 --eval 'console.table([\'Hello world\', `＂demo＂`])'
┌───┬─────────────┐
│   │ Values      │
├───┼─────────────┤
│ 0 │ Hello world │
│ 1 │ ＂demo＂     │
└───┴─────────────┘

~/Desktop/ok via 🍞 v1.1.34
❯ bun-1.1.7 --eval 'console.table([\'Hello world\', `＂demo＂`])'
┌───┬─────────────┐
│   │ Values      │
├───┼─────────────┤
│ 0 │ Hello world │
│ 1 │ ＂demo＂    │
└───┴─────────────┘

~/Desktop/ok via 🍞 v1.1.34
❯ bun --eval 'console.table([\'Hello world\', `＂demo＂`])'
┌───┬─────────────┐
│   │ Values      │
├───┼─────────────┤
│ 0 │ Hello world │
│ 1 │ ＂demo＂    │
```
